### PR TITLE
[MoE Refactor] Optional router_logits argument

### PR DESCRIPTION
**Pull request by bnellnm**
## Purpose

When `FusedMoE` handles the gate operation, the router logits do not need to be passed to the experts `forward` method.

- update models
- remove unused `is_internal_router` method
- update layer test

## Test Plan

CI
Ran deepseek v2 by hand.

## Test Result

@yzong-rh 

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: vllm/model_executor/layers/fused_moe/runner/moe_runner_interface.py
Comment (yzong-rh) [inline vllm/model_executor/layers/fused_moe/runner/moe_runner_interface.py]: Should probly be optional here too.

**Review by yzong-rh — APPROVED** [review]
LGTM as long as we are sure we don't need `is_internal_router` anymore.

**Comment by bnellnm** [comment]
> LGTM as long as we are sure we don't need `is_internal_router` anymore.

The only uses of it were in the models that I changed.

**Review by robertgshaw2-redhat — CHANGES_REQUESTED** [review]
If we now always run the gate internally, why do we need to accept router_logits as input at all?

Can't we just drop that?

**Comment by bnellnm** [comment]
> If we now always run the gate internally, why do we need to accept router_logits as input at all?
> 
> Can't we just drop that?

It's only run internally if the gate is passed as an argument to `FusedMoE`.  Not all models do this.

**Comment by bnellnm** [comment]
@robertgshaw2-redhat can you take another look at this?  I think I've resolved/explained the issue you had.

**Review by robertgshaw2-redhat — APPROVED** [review]

**Review by robertgshaw2-redhat — COMMENTED** [review]
test

**Review by robertgshaw2-redhat — COMMENTED** [review]
test

**Review by robertgshaw2-redhat — CHANGES_REQUESTED** [review]
i want to look at this again

**Review by hmellor — COMMENTED** [review]
The Transformers modelling backend part LTGM. If the Transformers modelling backend enters `_moe_block_forward` then the gate is always internal
